### PR TITLE
Add support to generate PWM info from DT/PWM LEDs

### PR DESCRIPTION
**Pull request by galak**
Added a series of patches to enable generation of pwm info based on device tree.  Also added for use of this based on pwm-leds on the hexiwear_k64.

### CODE REVIEW

File: dts/bindings/pwm/nxp,kinetis-ftm.yaml [hunk 1]
```diff
@@ -28,4 +28,9 @@ properties:
       category: required
       description: required interrupts
       generation: define
+
+"#cells":
+  - pin
```
Comment (galak) [inline dts/bindings/pwm/nxp,kinetis-ftm.yaml]: Wasn't sure if we want 'pin' or 'channel'?

Comment (MaureenHelm) [reply dts/bindings/pwm/nxp,kinetis-ftm.yaml]: channel

File: dts/arm/nxp/nxp_imx7d_m4.dtsi [hunk 2]
```diff
@@ -331,6 +331,7 @@
 					       RDC_DOMAIN_PERM_RW))>;
 			label = "PWM_1";
 			status = "disabled";
+			#pwm-cells = <2>;
```
Comment (MaureenHelm) [inline dts/arm/nxp/nxp_imx7d_m4.dtsi]: The imx changes should be in a separate commit

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
I like the direction, but a few changes are needed.

**Comment by MaureenHelm** [comment]
@diegosueiro please have a look

File: dts/arm/nxp/nxp_imx7d_m4.dtsi (hunk 2 above)
Comment (galak) [reply dts/arm/nxp/nxp_imx7d_m4.dtsi]: I was wondering where those went ;) [got merged accidentally in here]

Comment (galak) [reply dts/arm/nxp/nxp_imx7d_m4.dtsi]: Dropped the imx change for now

**Comment by galak** [comment]
Dropped the imx changes from this PR and renamed pin to channel in the binding.

File: samples/basic/blink_led/src/main.c [hunk 3]
```diff
@@ -29,18 +29,15 @@
...
-#define PWM_CHANNEL	GREEN_PWM_CHANNEL
 #elif defined(CONFIG_BOARD_COLIBRI_IMX7D_M4)
 #define PWM_DRIVER	PWM_1_LABEL
 #define PWM_CHANNEL	0
 #elif defined(CONFIG_SOC_FAMILY_NRF)
 #define PWM_DRIVER	CONFIG_PWM_NRF5_SW_0_DEV_NAME
 #define PWM_CHANNEL	LED0_GPIO_PIN
-#else
-#error "Choose supported PWM driver"
+#else /* get the defines from dt (based on alias 'pwm-led0') */
```
Comment (erwango) [inline samples/basic/blink_led/src/main.c]: What if PWM is not supported/not configured?
I guess for CI, we'll be able to get information from yaml/edts, but what about zephyr user that want to have a try?
Errors will be around PWM_LED0_PWM_CONTROLLER and PWM_LED0_PWM_CHANNEL not defined, which are generated so can't even grep on them... I think a new zephyr user could feel a bit lost ...

**Review by erwango — CHANGES_REQUESTED** [review]
Nice addition but I think we should keep proper error messages in samples

File: samples/basic/blink_led/src/main.c (hunk 3 above)
Comment (galak) [reply samples/basic/blink_led/src/main.c]: Put the error conditions back

**Review by erwango — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Comment by MaureenHelm** [comment]
Merging #10936 should fix the CI failure

File: dts/bindings/pwm/nxp,kinetis-ftm.yaml [hunk 4]
```diff
@@ -28,4 +28,9 @@ properties:
       category: required
       description: required interrupts
       generation: define
+
+"#cells":
```
Comment (b0661) [inline dts/bindings/pwm/nxp,kinetis-ftm.yaml]: #cells will  sometime in the future clash when a device integrates several cell related bindings (e.g #clock-cells -> #cells, #pwm-cells -> #cells). We should really start using some more specific here like #pwm-cells.

**Comment by MaureenHelm** [comment]
recheck

File: dts/bindings/pwm/nxp,kinetis-ftm.yaml (hunk 4 above)
Comment (galak) [reply dts/bindings/pwm/nxp,kinetis-ftm.yaml]: > #cells will sometime in the future clash when a device integrates several cell related bindings (e.g #clock-cells -> #cells, #pwm-cells -> #cells). We should really start using some more specific here like #pwm-cells.

Let's open that up as an issue.
